src/main/java/core/basesyntax/io/impl/CsvReportSerializer.java

```
@@ -0,0 +1,39 @@
+package core.basesyntax.io.impl;
+
+import core.basesyntax.io.ReportSerializer;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.model.Report;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvReportSerializer implements ReportSerializer {
+    @Override
+    public void save(Report report, Path path) {
+        List<String> csvRows = createCsvRows(report);
+        try (BufferedWriter writer = Files.newBufferedWriter(path)) {
+            for (String line : csvRows) {
+                writer.write(line);
+                writer.newLine();
+            }
+            writer.flush();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Catching IOException and rethrowing `new RuntimeException(e)` loses important context about what was being attempted and which path failed. The task requires: "Throw informative RuntimeExceptions on file read/write errors, including file path details. [CHECKLIST ITEM #Don't ignore exceptions in try-catch blocks]" Update the catch to include the `path` in the message, for example: `throw new RuntimeException("Unable to save report to " + path + ": " + e.getMessage(), e);`